src/test/java/com/crud/tasks/controller/TaskControllerTest.java

```diff
@@ -0,0 +1,127 @@
+
+@SpringJUnitWebConfig
+@WebMvcTest(TaskController.class)
+public class TaskControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private DbService dbService;
+
+    @MockBean
+    private TaskMapper taskMapper;
+
+    @Test
+    public void testGetTasks() throws Exception {
+        //Given
+        List<TaskDto> taskDtoList = List.of(new TaskDto(1L, "Test_title", "Content_test"));
+        when(taskMapper.mapToTaskDtoList(anyList())).thenReturn(taskDtoList);
+        //When & Then
+        mockMvc
+                .perform(MockMvcRequestBuilders
+                        .get("/v1/task/getTasks")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].title", Matchers.is("Test_title")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].content", Matchers.is("Content_test")));
+    }
+
+    @Test
+    public void testGetTaskById() throws Exception {
+        //Given
+        TaskDto taskDto = new TaskDto(1L, "title_test", "content_test");
+        Task task = new Task(1L, "title_test", "content_test");
+        when(dbService.getTaskById(anyLong())).thenReturn(Optional.of(task));
+        when(taskMapper.mapToTaskDto(any(Task.class))).thenReturn(taskDto);
+        //When & Then
+        mockMvc
+                .perform(MockMvcRequestBuilders
+                        .get("/v1/task/getTask/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .characterEncoding("UTF-8"))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title", Matchers.is("title_test")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.content", Matchers.is("content_test")));
+    }
+
+    @Test
+    public void testDeleteTaskById() throws Exception {
+        //Given
+        Task task = new Task(1L, "title_test", "content_test");
+        //When & Then
+        mockMvc
+                .perform(MockMvcRequestBuilders
+                        .delete("/v1/task/deleteTask/1")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().is(200));
+    }
+
+    @Test
+    public void testUpdateTask() throws Exception {
+        //Given
+        TaskDto taskDto = new TaskDto(1L, "title_test", "content_test");
+        Task task = new Task(1L, "title_test", "content_test");
+        when(taskMapper.mapToTask(any(TaskDto.class))).thenReturn(task);
+        when(dbService.saveTask(any(Task.class))).thenReturn(task);
+        when(taskMapper.mapToTaskDto(any(Task.class))).thenReturn(taskDto);
+
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(taskDto);
```

Comment:
Podpoiwedź: możesz zapoznać się z innymi (de)serializatorami JSONów. GSON to tylko jedno z dostępnych rozwiązań. Np. dobrze zintegrowany ze springiem jest `Jackson` , klasa to `ObjectMapper` (zobacz tu: https://www.baeldung.com/jackson-object-mapper-tutorial).
Nie mówię oczywiście, żeby zmieniać - tylko żebyś miał świadomość, że są różne biblioteki. Gdybyś wybierał taką bibliotekę do swojego projektu, na co zwróciłbyś uwagę?